### ISSUE TITLE
Initial review comments

### ISSUE THREAD

**Issue by mpadge** [ropensci-review-tools/pkgmatch#88]
https://github.com/ropensci/software-review/issues/671#issuecomment-2518775363 from @MargaretSiple-NOAA

> The pkgdown page seems to indicate that the user can set the corpus to rOpenSci or to CRAN, but some functions only provide search results for rOpenSci, e.g., pkgmatch_bm25() requires the corpus to be rOpenSci-related. Some clarity about which functions apply to which package corpi (?) might be helpful.

- [x] Update bm25 docs
- [x] Ensure clarity across all functions about corpus applicability

> I got 1 note from running devtools::check(): "Package suggested but not available for checking: 'piggyback'" -- I don't worry about these too much just wanted to flag it just in case.

- [x] Ensure piggyback properly suggested

@MargaretSiple-NOAA `devtools::check()` runs with a default argument, `force_suggests = FALSE`. The `piggyback` package is in `Suggests`, not `Imports`, so not required by default, but then if you don't actually have it locally installed, you'll see this error.

> The ollama connection gave me some trouble at the start, so I imagine future users might struggle on this. I recommend revising the documentation a little bit to make it more obvious what needs to happen with ollama before users can get started: a) Consider adding a sentence to the top of the "Get started" article that indicates that docker and ollama installs need to happen first and foremost. Here, I would link to the "ollama" article. b) Change the title of the "ollama" article to say something like "Before you begin: ollama" or "Before you begin: ollama installation", so that when people are browsing, it's very clear to them that they'll need to read that article first.

- [x] Rename title of ollama article
- [x] Restructure beginning of article

> The beginning of the [Docker section](https://docs.ropensci.org/pkgmatch/articles/ollama.html#docker) in the ollama article should indicate that users need to have Docker installed as well. I learned from this process that my institution actually doesn't have a Docker license! This may impede editors/reviewers who work at NOAA. It sounds like they're working on getting a license so that may change in the future, so that will probably change in the future.

- [x] Clarify that ollama installation generally doesn't require Docker, and that Docker is an alternative approach to "standard" install
- [x] Clarify that a Docker installation is required for that to work

> I had some trouble getting test coverage my usual way (covr::package_coverage() fails with an error about a temp file)... but devtools::test() works great). Some tests are failing but I'm pretty sure that's because I don't have ollama properly installed on my machine. It might be nice to have a one-line check in one of the early vignettes that can show people whether they are missing any of the components need to run the fns.

- [x] Investigate `covr` vs `devtools` issue

I also see an error, but in my case it's because of https://github.com/r-lib/covr/issues/537, which hasn't yet been resolved.  But checking in a clean docker container (`rocker/tidyverse`) all works fine, 

- [x] Consider clarifying somewhere that test suite is entirely mocked and doesn't require ollama at all

Added [`tests/README.md`](https://github.com/ropensci-review-tools/pkgmatch/tree/main/tests) explaining that point, and clarifying a couple of coverage exceptions.
